src/components/DataSession/OperationPipeline.vue

```diff
@@ -56,20 +56,32 @@ function operationBtnColor(index) {
 async function pollOperationCompletion(operationID) {
   const url = store.datalabApiBaseUrl + 'datasessions/' + props.session_id + '/operations/' + operationID + '/'
 
-  const updateOperationPercentages = (response) => {
-    if (response && response.percent_completion !== undefined && response.status != 'COMPLETED') {
-      operationPercentages.value[operationID] = response.percent_completion * DEC_TO_PERCENT
-    } else {
-      operationPercentages.value[operationID] = COMPLETE_PERCENT
-      emit('operationCompleted', response)
-      if (operationID in operationPollingTimers) {
-        clearInterval(operationPollingTimers[operationID])
-        setTimeout(clearPollingData(operationID), POLL_WAIT_TIME)
+  const updateOperationStatus = (response) => {
+    if(response){
+      switch(response.status){
+      case 'PENDING':
+        break
+      case 'IN_PROGRESS':
+        operationPercentages.value[operationID] = response.percent_completion * DEC_TO_PERCENT
+        break
+      case 'COMPLETED':
+        operationPercentages.value[operationID] = COMPLETE_PERCENT
+        emit('operationCompleted', response)
+        if (operationID in operationPollingTimers) {
+          clearInterval(operationPollingTimers[operationID])
+          setTimeout(clearPollingData(operationID), POLL_WAIT_TIME)
+        }
+        break
+      default:
+        console.error('Unknown Operation Status:', response.status)
       }
     }
+    else{
+      console.error('No response on status for operation:', operationID)
+    }
   }
 
-  await fetchApiCall({ url: url, method: 'GET', successCallback: updateOperationPercentages, failCallback: handleError })
+  await fetchApiCall({ url: url, method: 'GET', successCallback: updateOperationStatus, failCallback: handleError })
```

Comment:
So maybe here you can do updateOperationStatus(operationID, response). Something along those lines

Reply:
my mind completely missed this option...